CMakeLists.txt

```diff
@@ -252,6 +252,16 @@ if(BUILD_TESTING)
     if (NOT TARGET GTest::gtest)
       message(STATUS "Defining GTest::gtest alias to work-around bug in older release.")
       add_library(GTest::gtest ALIAS gtest)
+
+      # NOTE: gtest uninit some variables, gcc >= 1.11.3 may cause error on compile.
+      # Remove this comment and six lines below, once ninja deps gtest-1.12.0 or above.
+      if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "1.11.3")
+        check_cxx_compiler_flag(-Wno-maybe-uninitialized flag_no_maybe_uninit)
+        if (flag_no_maybe_uninit)
+          target_compile_options(gtest PRIVATE -Wno-maybe-uninitialized)
+        endif()
+      endif()
```

Comment:
Can you put this section after the endif() for the GTest::gtest check? Otherwise it will not be interpreted when compiling against a version of GTest that is more recent than 1.10 (such as 1.11 which would still need this)

Reply:
actually, 1.11 is fixed. see [this pr]( https://github.com/google/googletest/pull/3024/commits/4679637f1c9d5a0728bdc347a531737fad0b1ca3)

Reply:
Ah, good to know. In this case please update the comment to reference 1.11 instead of 1.12 :)

Reply:
sure, it's ok now